### ISSUE TITLE
pointfeat_1 meaning

### ISSUE THREAD

**Issue by trevor-taeyeop** [j96w/DenseFusion#56]
> Hi, a _PointNet-based_ network that processes each point in the masked 3D point cloud to a geometric feature embedding, said in the paper. But in the code implementation only convolution and ReLU of point cloud _x_ in _network.py_. No spacial transform networks(STN) and maxpooling operation on point cloud. These are the two most important features of PointNet.
> 
> ` Class PoseNetFeat (nn. Module):
> def **init**(self, num_points):
> super(PoseNetFeat, self).**init**()
> self.conv1 = torch.nn.Conv1d(3, 64, 1)
> self.conv2 = torch.nn.Conv1d(64, 128, 1)
> 
> ```
>     self.e_conv1 = torch.nn.Conv1d(32, 64, 1)
>     self.e_conv2 = torch.nn.Conv1d(64, 128, 1)
> 
>     self.conv5 = torch.nn.Conv1d(256, 512, 1)
>     self.conv6 = torch.nn.Conv1d(512, 1024, 1)
> 
>     self.ap1 = torch.nn.AvgPool1d(num_points)
>     self.num_points = num_points
> def forward(self, x, emb):
>     x = F.relu(self.conv1(x))
>     emb = F.relu(self.e_conv1(emb))
>     pointfeat_1 = torch.cat((x, emb), dim=1)
> 
>     x = F.relu(self.conv2(x))
>     emb = F.relu(self.e_conv2(emb))
>     pointfeat_2 = torch.cat((x, emb), dim=1)
> 
>     x = F.relu(self.conv5(pointfeat_2))
>     x = F.relu(self.conv6(x))
> 
>     ap_x = self.ap1(x)
> 
>     ap_x = ap_x.view(-1, 1024, 1).repeat(1, 1, self.num_points)
>     return torch.cat([pointfeat_1, pointfeat_2, ap_x], 1) #128 + 256 + 1024`
> ```
> 
> I am confused. Would you explain this to me?
> Thanks in advance.

I have some related question about this part.

1. I'm confused in this code which variable represents color embeddings and geometry embeddings??
 I understood pointfeat_2(color embeding+geometry embeding), ap_x(global_feature) based on the paper embeding dimensions are 128 . 

what's the meaning of        
>     pointfeat_1 = torch.cat((x, emb), dim=1)

>     return torch.cat([pointfeat_1, pointfeat_2, ap_x], 1) #128 + 256 + 1024`

Would you explain about this part?? 
Thanks.

_Originally posted by @trevor-taeyeop in https://github.com/j96w/DenseFusion/issues/34#issuecomment-503521928_

**Comment by j96w** [comment]
Hi, the pointfeat_1 is just an early fusion step of the two embeddings.
